### ISSUE TITLE
Collection expressions missing some nullability analysis

### ISSUE THREAD

**Issue by hamarb123** [dotnet/roslyn#76929]
**Version Used**: sharplab.io (also reproduces in my VS installation version 17.12.0, but this is probably not up to date lol)

**Steps to Reproduce**:

```csharp
using System;
using System.Collections.Generic;
public class C {
    public static void Main() {
        List<int>? a = null;
        List<int> b = [.. a]; //this should warn that a might be null
    }
}
```

[Link](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgFgAoIgBgAIiyUBuaogZkZPoGF6A3tXojuPIVVFT6AGQCWAZwwAeOXgwA+APz0AhvQC89PDgA2ptpOkj5S1eo31gh+gG0AdO70BdFvQD0/hgAFor0CsEQZgAm9ADuulB49CG6GHr0ALZyAObB6cAwxmamwqIAvtTlQA==)

**Expected Behavior**:

The collection expression `[.. a]` should warn that `a` might be `null`.

**Actual Behavior**:

No warning is emitted, and an exception is thrown at runtime for the sample above.

**Comment by RikkiGibson** [comment]
SharpLab is in a wonky state with Roslyn main right now, but, it looks like this scenario was fixed by #74686.

At the moment using a recent-ish feature branch seems to make the sample work as expected: [SharpLab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDs4BsuoMYD26AJgJYylFS5TLKkDm6AthOjAD4ACATAIwBYAFBcADAAIufACwBuYVwDMknuIDC4gN7Dxulau1C9x8QBlSyGAB5S7AHwB+cVHEBecZhzyjJ3ecs29uLAbuIA2gB0Ec4AurLiAPQJMAAWFuLIKQTYxOIA7lAI6OKpsM7izIwpMMEQHthYOnoAvsLNQA===)
